### PR TITLE
BRB 181 FWU routing

### PR DESCRIPTION
**Pull request by SteKrause**
# Description
A feature flag(FEATURE_FWU_CONTENT_ENABLED) for the FWU routing in the server has been added.
Added additional ingress path for the microservice which runs fwu-learning-contents module.
Microservice is only available on brb-instance.
Secrets and configurations for the S3 ('FWU_CONTENT__') are set via 1Password/dof_app_deploy.

## Links to Tickets or other pull requests
Ticket:
https://ticketsystem.dbildungscloud.de/browse/BRB-223

Epic:
https://ticketsystem.dbildungscloud.de/browse/BRB-181

Initiative:
https://ticketsystem.dbildungscloud.de/browse/BRB-178

PR schulcloud server:
https://github.com/hpi-schul-cloud/schulcloud-server/pull/3975

PR docker-compose:
https://github.com/hpi-schul-cloud/docker-compose/pull/62

## Changes
FEATURE_FWU_CONTENT_ENABLED has been added.
Added configurations for the S3 with FWU contents.
Added new ingress path for the microservice.

## Deployment
FEATURE_FWU_CONTENT_ENABLED, the default value is false.
It is only enabled (true) for BRB, because they got the licenses for FWU.

## Approval for review

- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

**Review by virgilchiriac — APPROVED** [review]

**Comment by virgilchiriac** [comment]
pls do not merge without the server PR (wait for approval for that one, and then merge at the same time)

File: ansible/group_vars/all/config.yml [hunk 1]
```diff
@@ -561,6 +561,11 @@ configuration_all:
     server: true
     client: false
     nuxtclient: false
+  FEATURE_FWU_CONTENT_ENABLED:
+    value: "false"
+    server: true
+    client: false
+    nuxtclient: false
```
Comment (virgilchiriac) [inline ansible/group_vars/all/config.yml:568]: what about the other configuration FWU_CONTENT__ ?

**Review by virgilchiriac — CHANGES_REQUESTED** [review]

File: ansible/group_vars/all/config.yml (hunk 1 above)
Comment (SteKrause) [reply ansible/group_vars/all/config.yml:568]: The configuration for 'FWU_CONTENT__' takes place via 1Password.

File: ansible/group_vars/all/x_ingress.yml [hunk 2]
```diff
@@ -1,4 +1,5 @@
-default_ingress:
+default_ingress: "{{ default_ingress_rules | combine( additional_ingress_rules | default({})) }}"
```
Comment (Loki-Afro) [inline ansible/group_vars/all/x_ingress.yml:1]: there is already something called `group_ingress` why not use that?

File: ansible/host_vars/prod-brb/cfg.yml [hunk 3]
```diff
@@ -5,6 +5,9 @@ FEATURE_NEXBOARD_ENABLED: "true"
 
 ALERT_STATUS_URL: https://status.brandenburg.cloud
 
+# FWU_CONTENT__S3_BUCKET: fwu-content-prod
```
Comment (Loki-Afro) [inline ansible/host_vars/prod-brb/cfg.yml:8]: remove commented out stuff

File: ansible/host_vars/ref-brb/cfg.yml [hunk 4]
```diff
@@ -7,6 +7,9 @@ FEATURE_SKIP_FIRST_LOGIN_ENABLED: "true"
 DISPLAY_REQUEST_LEVEL: -1
 ALERT_STATUS_URL: https://status.brandenburg.cloud
 
+# FWU_CONTENT__S3_BUCKET: fwu-content-staging
```
Comment (Loki-Afro) [inline ansible/host_vars/ref-brb/cfg.yml:10]: remove commented out stuff

**Review by Loki-Afro — CHANGES_REQUESTED** [review]

**Review by Loki-Afro — APPROVED** [review]

File: ansible/host_vars/prod-brb/cfg.yml (hunk 3 above)
Comment (SteKrause) [reply ansible/host_vars/prod-brb/cfg.yml]: removed comments

File: ansible/host_vars/ref-brb/cfg.yml (hunk 4 above)
Comment (SteKrause) [reply ansible/host_vars/ref-brb/cfg.yml]: removed comments

File: ansible/group_vars/all/config.yml [hunk 5]
```diff
@@ -572,6 +572,30 @@ configuration_all:
...
+    value: "http://localhost:4568"
+    server: true
+    client: false
+    nuxtclient: false
+  FWU_CONTENT__S3_BUCKET:
+    value: ""
+    server: true
+    client: false
+    nuxtclient: false
+  FWU_CONTENT__INCOMING_REQUEST_TIMEOUT_MS:
```
Comment (virgilchiriac) [inline ansible/group_vars/all/config.yml:595]: should this not have a value prop?

**Review by virgilchiriac — APPROVED** [review]

File: ansible/group_vars/all/config.yml (hunk 5 above)
Comment (SteKrause) [reply ansible/group_vars/all/config.yml:595]: There's always a default value in the server for the timeout.

**Pull request merged**